### ISSUE TITLE
Fails to build on NixOS

### ISSUE THREAD

**Issue by Soliprem** [KZDKM/Hyprspace#60]
This is the error that I get from running ```home-manager switch --flake <flake/path>#user@hostname```

```error: builder for '/nix/store/3hk2g9g52v1d85d0jy4wpd9dmxfb9b30-Hyprspace-0.1+date=2024-06-17_2f3edb6.drv' failed with exit code 1;
       last 10 log lines:
       >       |                        getWorkspaceIDNameFromString
       > [2/6] Compiling C++ object libHyprspace.so.p/src_Layout.cpp.o
       > [3/6] Compiling C++ object libHyprspace.so.p/src_Overview.cpp.o
       > [4/6] Compiling C++ object libHyprspace.so.p/src_Render.cpp.o
       > ../src/Render.cpp: In member function 'void CHyprspaceWidget::draw()':
       > ../src/Render.cpp:117:247: warning: narrowing conversion of 'Config::panelBorderWidth' from 'int' to 'double' [-Wnarrowing]
       >   117 |         CBox borderBox = {widgetBox.x, owner->vecPosition.y + (Config::onBottom * owner->vecTransformedSize.y) + (Config::panelHeight + Config::reservedArea - curYOffset.value()* owner->scale) * bottomInvert, owner->vecTransformedSize.x, Config::panelBorderWidth};
       >       |                                                                                                                                                                                                                                               ~~~~~~~~^~~~~~~~~~~~~~~~
       > [5/6] Compiling C++ object libHyprspace.so.p/src_main.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/3hk2g9g52v1d85d0jy4wpd9dmxfb9b30-Hyprspace-0.1+date=2024-06-17_2f3edb6.drv'.
error: 1 dependencies of derivation '/nix/store/gch402j4gg15jcc88jg79w8z37mg04y3-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ip0p4gnp30z7l5d4569ajf5igvr9lh66-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1jfyxni24acgz3ir27rim8ssvz155pp7-home-manager-generation.drv' failed to build```

**Comment by cootshk** [comment]
I'm getting the same error trying to build hyprspace

**Comment by cootshk** [comment]
my full log:
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/8gq8q6knn3iv9705qb49h0g1aqfyxz2s-ngp037bikjcnvzy284fl9ncla43m3l1f-source
source root is ngp037bikjcnvzy284fl9ncla43m3l1f-source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6 --libdir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/lib --libexecdir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/libexec --bindir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/bin --sbindir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/sbin --includedir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/include --mandir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/man --infodir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/info --localedir=/nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain
[1mThe Meson build system[0m
Version: 1.4.0
Source dir: [1m/build/ngp037bikjcnvzy284fl9ncla43m3l1f-source[0m
Build dir: [1m/build/ngp037bikjcnvzy284fl9ncla43m3l1f-source/build[0m
Build type: [1mnative build[0m
Project name: [1mHyprspace[0m
Project version: [1m0.1[0m
C++ compiler for the host machine: [1mg++[0m (gcc 13.3.0 "g++ (GCC) 13.3.0")
C++ linker for the host machine: [1mg++[0m ld.bfd 2.41
Host machine cpu family: [1mx86_64[0m
Host machine cpu: [1mx86_64[0m
Compiler for C++ supports arguments -std=c++23: [1;32mYES[0m 
meson.build:8: [1;33mWARNING:[0m Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: [1;32mYES[0m [1m(/nix/store/2hfzaqv42iwrpk3ya24cnjsklz6f68lw-pkg-config-wrapper-0.29.2/bin/pkg-config)[0m [1;34m0.29.2[0m
Run-time dependency [1mhyprland[0m found: [1;32mYES[0m [36m0.41.2[0m
Run-time dependency [1mpixman-1[0m found: [1;32mYES[0m [36m0.43.4[0m
Run-time dependency [1mlibdrm[0m found: [1;32mYES[0m [36m2.4.120[0m
Build targets in project: [1m1[0m

Hyprspace [36m0.1[0m

  [1mUser defined options[0m
    auto_features: enabled
    bindir       : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/bin
    buildtype    : plain
    includedir   : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/include
    infodir      : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/info
    libdir       : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/lib
    libexecdir   : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/libexec
    localedir    : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/locale
    mandir       : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/share/man
    prefix       : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6
    sbindir      : /nix/store/7wcv642rbycbdr074lycl2bqn29qlk5q-Hyprspace-0.1+date=2024-06-17_2f3edb6/sbin
    wrap_mode    : nodownload

Found ninja-1.12.1 at /nix/store/v3wh3z7j8dxlb3306ackf25h8hw7lzg2-ninja-1.12.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j24
[1/6] Compiling C++ object libHyprspace.so.p/src_Input.cpp.o
FAILED: libHyprspace.so.p/src_Input.cpp.o 
g++ -IlibHyprspace.so.p -I. -I.. -I/nix/store/wqfar0hyz565qq0c1i1qarpihr96f9lg-hyprland-0.41.2+date=2024-06-27_b7f42a1-dev/include -I/nix/store/wqfar0hyz565qq0c1i1qarpihr96f9lg-hyprland-0.41.2+date=2024-06-27_b7f42a1-dev/include/hyprland/protocols -I/nix/store/wqfar0hyz565qq0c1i1qarpihr96f9lg-hyprland-0.41.2+date=2024-06-27_b7f42a1-dev/include/hyprland -I/nix/store/wqfar0hyz565qq0c1i1qarpihr96f9lg-hyprland-0.41.2+date=2024-06-27_b7f42a1-dev/include/hyprland/wlr -I/nix/store/avzjkq3q95vjisfpx018ff8l9zlxn9hf-pixman-0.43.4/include/pixman-1 -I/nix/store/5f2704za85mzq80m18pb6bc7fy77bzsg-libdrm-2.4.120-dev/include -I/nix/store/5f2704za85mzq80m18pb6bc7fy77bzsg-libdrm-2.4.120-dev/include/libdrm -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -DWLR_USE_UNSTABLE -std=c++23 -fPIC -MD -MQ libHyprspace.so.p/src_Input.cpp.o -MF libHyprspace.so.p/src_Input.cpp.o.d -o libHyprspace.so.p/src_Input.cpp.o -c ../src/Input.cpp
../src/Input.cpp: In member function 'bool CHyprspaceWidget::axisEvent(double, Hyprutils::Math::Vector2D)':
../src/Input.cpp:92:24: error: 'getWorkspaceIDFromString' was not declared in this scope; did you mean 'getWorkspaceIDNameFromString'?
   92 |             int wsID = getWorkspaceIDFromString("r-1", outName);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~
      |                        getWorkspaceIDNameFromString
../src/Input.cpp:98:24: error: 'getWorkspaceIDFromString' was not declared in this scope; did you mean 'getWorkspaceIDNameFromString'?
   98 |             int wsID = getWorkspaceIDFromString("r+1", outName);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~
      |                        getWorkspaceIDNameFromString
[2/6] Compiling C++ object libHyprspace.so.p/src_Render.cpp.o
../src/Render.cpp: In member function 'void CHyprspaceWidget::draw()':
../src/Render.cpp:117:247: warning: narrowing conversion of 'Config::panelBorderWidth' from 'int' to 'double' [-Wnarrowing]
  117 |         CBox borderBox = {widgetBox.x, owner->vecPosition.y + (Config::onBottom * owner->vecTransformedSize.y) + (Config::panelHeight + Config::reservedArea - curYOffset.value()* owner->scale) * bottomInvert, owner->vecTransformedSize.x, Config::panelBorderWidth};
      |                                                                                                                                                                                                                                               ~~~~~~~~^~~~~~~~~~~~~~~~
[3/6] Compiling C++ object libHyprspace.so.p/src_Overview.cpp.o
[4/6] Compiling C++ object libHyprspace.so.p/src_Layout.cpp.o
[5/6] Compiling C++ object libHyprspace.so.p/src_main.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by Soliprem** [comment]
gets fixed by #62

**Comment by Soliprem** [comment]
Fixed by #62 being merged
